src/routes/user/updateUser.ts

```diff
@@ -0,0 +1,91 @@
+import { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';
+import { knexController } from '../../database/database';
+import { hashPassword } from '../../lib/auth';
+import {
+  User,
+  editUserResponse,
+  user,
+  UserParamsPayload,
+  HashObj,
+} from '../../types/user.types';
+
+const editUserSchema = {
+  description: 'Edit data of already existing user.',
+  summary: 'Edit user',
+  tags: ['User', 'Update'],
+  body: {
+    type: 'object',
+    required: [],
+    properties: user.static,
+  },
+  response: {
+    200: editUserResponse,
+    500: { $ref: 'error' },
+    400: { $ref: 'error' },
+    404: { $ref: 'error' },
+  },
+};
+
+const editUserHandler = async (
+  req: FastifyRequest<{ Body: User; Params: UserParamsPayload }>,
+  reply: FastifyReply
+): Promise<void> => {
+  const { email, password, forename, surname, isAdmin, isBlender } = req.body;
+
+  const userId = req.params.userId;
+
+  let hashObj: HashObj = {
+    hash: undefined,
+    salt: undefined,
+  };
+  // If no password given as parameter, no need to hash.
+  if (password !== undefined) {
+    hashObj = await hashPassword(password);
+  }
+
+  // edit user
+  const editResponse = await knexController('user')
+    .where({ id: userId })
+    .update({
+      email,
+      forename,
+      surname,
+      password_hash: hashObj.hash,
+      salt: hashObj.salt,
+      is_admin: isAdmin,
+      is_blender: isBlender,
+    });
+
+  // If no user found with given id.
+  if (editResponse === 0) {
+    return reply.code(404).send({
+      statusCode: 404,
+      error: 'Bad Request',
```

Comment:
Error would be `Not Found`.

https://developer.mozilla.org/en-US/docs/Web/HTTP/Status